lib/movie_reserver.rb

```diff
@@ -1,7 +1,121 @@
 require 'csv'
+require 'pry'
+require_relative 'date_range'
+require_relative 'movie'
+require_relative 'rental'
+
 
 module GreenBox
   class MovieReserver
+    attr_accessor :movies, :rentals, :date_range
+    # :reserver, :date_range
+
+    def initialize
+      @movies = MovieReserver.load_movies
+      @rentals = []
+      @date_range = GreenBox::DateRange.new(Time.parse('2018-08-08'),Time.parse('2018-08-09'))
+    end
+
+      # `self.load_movies`: This method will open the csv file `movies.csv` and read in the movies and return an array of the given movies.  Note the actors are separated by the `:` character.  You will need to break up that field.
+    def self.load_movies
+      showtime_movies  = []
+
+      CSV.read('data/movies.csv', headers: false).each do |line|
+        movie_id = line[0].to_i
+        title = line[1]
+        publisher = line[2]
+        all_actors = line[3]
+        actors_names = all_actors.split(':')
+        actors = {actors: actors_names}
+
+        showtime_movies << GreenBox::Movie.new(movie_id, title, publisher, actors)
+      end
+      return showtime_movies
+    end
+
+
+    # `available_movies(date_range)`:  This method will take a `DateRange` instance and return a list of
+    #  movies available (not rented yet) in that range.
+    def available_movies(date_range)
+
+      # @movies = available_movies
+      # If @rentals = [] - is empty,
+        if @rentals.empty?
+          # then all the movies are available
+          return @movies
+        else
+          # @movies is a MovieReserver object that contains movies.
+          available_movies = []
+          @movies.each do |movie|
+            if @rentals.date_range == date_range
```

Comment:
This part is not quite right.  `@rentals` is an array and so doesn't have a `date_range`  attribute.  
Instead you should:
1.  make a copy of the `movies` list
1.  then loop through your `rentals` array and remove any movies who have a rental that conflicts.  

You can look up the [`select`](https://ruby-doc.org/core-2.2.0/Array.html#method-i-select) method to find movies that don't have a given ID or object id.